src/plugins/plugin-http.ts

```diff
@@ -179,8 +183,14 @@ function makeRequestTrace(
     // Inject the trace context header, but only if it wasn't already injected
     // earlier.
     if (!traceHeaderPreinjected) {
-      req.setHeader(
-          api.constants.TRACE_CONTEXT_HEADER_NAME, span.getTraceContext());
+      try {
+        req.setHeader(
+            api.constants.TRACE_CONTEXT_HEADER_NAME, span.getTraceContext());
+      } catch (e) {
```

Comment:
Can you make this error a bit more specific rather than swallowing all errors? For example, swallow the error only if `e.code` matches your expectation.